applications/FluidDynamicsApplication/custom_elements/two_fluid_navier_stokes.cpp

```diff
@@ -1937,6 +1989,220 @@ void TwoFluidNavierStokes<TElementData>::ComputeSplitting(
     rData.NumberOfDivisions = (p_modified_sh_func->pGetSplittingUtil())->mDivisionsNumber;
 }
 
+template <class TElementData>
+void TwoFluidNavierStokes<TElementData>::ComputeSplitInterface(
+        TElementData &rData,
+        MatrixType& rInterfaceShapeFunctionNeg,
+        MatrixType& rEnrInterfaceShapeFunctionPos,
+        MatrixType& rEnrInterfaceShapeFunctionNeg,
+        GeometryType::ShapeFunctionsGradientsType& rInterfaceShapeDerivativesNeg,
+        Kratos::Vector& rInterfaceWeightsNeg,
+        std::vector<Vector>& rInterfaceNormalsNeg)
+{
+    Matrix enr_neg_interp = ZeroMatrix(NumNodes, NumNodes);
+    Matrix enr_pos_interp = ZeroMatrix(NumNodes, NumNodes);
+
+    for (unsigned int i = 0; i < NumNodes; i++){
+        if (rData.Distance[i] > 0.0){
+            enr_neg_interp(i, i) = 1.0;
+        } else{
+            enr_pos_interp(i, i) = 1.0;
+        }
+    }
+
+    // Construct the modified shape fucntions utility
+    GeometryType::Pointer p_geom = this->pGetGeometry();
+    ModifiedShapeFunctions::Pointer p_modified_sh_func = nullptr;
+    if (Dim == 2)
+        p_modified_sh_func = Kratos::make_shared<Triangle2D3ModifiedShapeFunctions>(p_geom, rData.Distance);
+    else
+        p_modified_sh_func = Kratos::make_shared<Tetrahedra3D4ModifiedShapeFunctions>(p_geom, rData.Distance);
```

Comment:
This is used already, I would refactor that to a template function so no if checks are performed inside `CalculateLocalSystem`

Reply:
done.